cg/apps/sequencing_metrics_parser/parsers/bcl_convert_to_sequencing_statistics.py

```diff
@@ -18,26 +19,55 @@ def create_sample_lane_sequencing_metrics_from_bcl_convert_metrics_for_flow_cell
     )
     sample_internal_ids: List[str] = metrics_parser.get_sample_internal_ids()
     sample_lane_sequencing_metrics: List[SampleLaneSequencingMetrics] = []
+
     for sample_internal_id in sample_internal_ids:
         for lane in metrics_parser.get_lanes_for_sample(sample_internal_id=sample_internal_id):
-            sample_lane_sequencing_metrics.append(
-                SampleLaneSequencingMetrics(
-                    sample_internal_id=sample_internal_id,
-                    flow_cell_name=FlowCellDirectoryData(
-                        flow_cell_path=flow_cell_dir, bcl_converter=BclConverter.BCLCONVERT
-                    ).id,
-                    flow_cell_lane_number=lane,
-                    sample_total_reads_in_lane=metrics_parser.calculate_total_reads_for_sample_in_lane(
-                        sample_internal_id=sample_internal_id, lane=lane
-                    ),
-                    sample_base_percentage_passing_q30=metrics_parser.get_q30_bases_percent_for_sample_in_lane(
-                        sample_internal_id=sample_internal_id, lane=lane
-                    ),
-                    sample_base_mean_quality_score=metrics_parser.get_mean_quality_score_for_sample_in_lane(
-                        sample_internal_id=sample_internal_id, lane=lane
-                    ),
-                    created_at=datetime.now(),
-                )
+            metrics: SampleLaneSequencingMetrics = create_sequencing_metrics(
+                sample_internal_id=sample_internal_id, lane=lane, metrics_parser=metrics_parser
             )
-
+            sample_lane_sequencing_metrics.append(metrics)
     return sample_lane_sequencing_metrics
+
+
+def create_bcl_convert_undetermined_metrics(
+    flow_cell_dir: Path, non_pooled_lane_sample_pairs: List[Tuple[int, str]]
+) -> List[SampleLaneSequencingMetrics]:
+    """Return sequencing metrics for any undetermined reads in the specified lanes."""
+    metrics_parser: BclConvertMetricsParser = BclConvertMetricsParser(flow_cell_dir)
+    undetermined_metrics: List[SampleLaneSequencingMetrics] = []
+
+    for lane, sample_internal_id in non_pooled_lane_sample_pairs:
+        if not metrics_parser.has_undetermined_reads_in_lane(lane):
+            continue
+        metrics: SampleLaneSequencingMetrics = create_sequencing_metrics(
+            sample_internal_id=UNDETERMINED, lane=lane, metrics_parser=metrics_parser
```

Comment:
Should we not store it under the actual 'sample_internal_id' that these read will be assigned to? Or how are going to link those reads to the sample they belong to?

Reply:
Yes, that is true! The sample id is updated below.
`UNDETERMINED` is passed here since we want to create metrics for reads tagged as such in the bcl convert output.

Reply:
Added a comment to clarify this. Could create a separate function which encapsulates this like
```python
def create_undetermined_bcl_convert_sequencing_metrics(
    sample_internal_id: str, lane: int, 
    metrics_parser: BclConvertMetricsParser
):
    # Call create_bcl_convert_sequencing_metrics with Undetermined as sample id
    # Update sample id on metrics
```
Would that be even clearer?

Reply:
Ah I see, I was just wondering why we would do this in two steps? We could just set the sample internal ID when we initialise the metrics here instead of assigning it below.
```suggestion
            sample_internal_id=sample_internal_id, lane=lane, metrics_parser=metrics_parser
```